nexus/src/app/instance.rs

```diff
@@ -712,25 +706,13 @@ impl super::Nexus {
     pub async fn instance_attach_disk(
         &self,
         opctx: &OpContext,
-        organization_name: &Name,
-        project_name: &Name,
-        instance_name: &Name,
-        disk_name: &Name,
+        instance_lookup: &lookup::Instance<'_>,
+        disk_lookup: &lookup::Disk<'_>,
     ) -> UpdateResult<db::model::Disk> {
-        let (.., authz_project, authz_disk, _) =
-            LookupPath::new(opctx, &self.db_datastore)
-                .organization_name(organization_name)
-                .project_name(project_name)
-                .disk_name(disk_name)
-                .fetch()
-                .await?;
-        let (.., authz_instance, _) =
-            LookupPath::new(opctx, &self.db_datastore)
-                .project_id(authz_project.id())
-                .instance_name(instance_name)
-                .fetch()
-                .await?;
-
+        let (.., authz_instance) =
```

Comment:
What happens if the disk is in a different Project?  (Also, do we have a test for that?)

Before, it wasn't possible to specify this.  Now it seems like it is.

Reply:
I've updated the code here. It now uses a lookup for the instance and gets the `authz_project` then does a lookup for the disk with the project id as the root of the lookup. I think what you're saying still may apply though. If you try to attach a disk by UUID then the project id in the lookup would be ignored and the disk UUID would be used instead. I'll add a check for this, good catch.

Reply:
Fix is added in [8f79dc6](https://github.com/oxidecomputer/omicron/pull/2008/commits/8f79dc630002f082eab9f655e2a03a194ebe52ce). I only added it to the attach case because the detach case only operates on whats already attached so it's an unnecessary check there. In the error message I noticed that they must be in the same project. This technically tells the user something about the relationship of the resources they may not have known, but I think that's fine because the step before should ensure they have modify permissions on both resources.

Reply:
Great.  I'm not sure what the plan is for TODO-v1 but this seems like a test we want sooner rather than later because it feels easy to break without noticing and it seems really bad if we accidentally ship it that way.

Reply:
All the `TODO-v1` comments will be addressed before switching to the v1 endpoints.